### PR TITLE
Import zdfs package to support a tricky OCIv1 tgz format

### PR DESCRIPTION
**Pull request by BigVan**
**What this PR does / why we need it**:
Add zdfs package import.
Packing the reference files of the overlaybd blob into the layer of the OCIv1 image; This enables the old version of containerd or other runtime (such as docker) to run overlaybd image

reference files like:
```bash
ls /var/lib/containerd/io.containerd.snapshotter.v1.overlaybd/snapshots/283/fs -a
.aaaaaaaaaaaaaaaa.lsmt  .checksum_file  .data_size  .oss_url  .type

# .oss_url is the overlaybd blob uploaded address in remote storage
# .data_size is the blobSize record in the annotation of overlaybd image manifest.
# other files can be empty
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: pkg/snapshot/overlay.go [hunk 1]
```diff
@@ -1160,6 +1169,21 @@ func (o *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 		}
 	}
 
+	_, info, _, err := storage.GetInfo(ctx, key)
```
Comment (liulanzheng) [inline pkg/snapshot/overlay.go:1172]: check this err?

Comment (BigVan) [reply pkg/snapshot/overlay.go:1172]: I think 'image_ref' can be saved after info get in Line 1193?
image

**Comment by liulanzheng** [comment]
LGTM  @WaberZhuang  cc

File: pkg/snapshot/overlay.go [hunk 2]
```diff
@@ -1170,6 +1179,18 @@ func (o *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
...
 		return "", snapshots.Info{}, errors.Wrap(err, "failed to get snapshot info")
 	}
+	img, ok := info.Labels[label.CRIImageRef]
+	if !ok {
+		img, ok = info.Labels[label.TargetImageRef]
+	}
+	if ok {
+		log.G(ctx).Infof("found imageRef: %s", img)
+		if err := os.WriteFile(filepath.Join(td, "image_ref"), []byte(img), 0644); err != nil {
+			log.G(ctx).Errorf("LSMD ERROR write imageRef '%s'. path: %s, err: %s", img, filepath.Join(td, "image_ref"), err.Error())
```
Comment (WaberZhuang) [inline pkg/snapshot/overlay.go:1189]: nit: `LSMD ERROR` in the log is confusing, as is `err.Error()`

**Review by WaberZhuang — CHANGES_REQUESTED** [review]

File: pkg/snapshot/overlay.go [hunk 3]
```diff
@@ -777,6 +779,13 @@ func (o *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 		}
 	}
 
+	if isOverlaybd, err := zdfs.PrepareOverlayBDSpec(ctx, key, id, o.snPath(id), oinfo, o.snPath); isOverlaybd {
+		log.G(ctx).Infof("sn: %s is an overlaybd layer", id)
+	} else if err != nil {
+		log.G(ctx).Errorf("prepare overlaybd spec failed(sn: %s): %s", id, err.Error())
```
Comment (liulanzheng) [inline pkg/snapshot/overlay.go:785]: use ("%v", err) to print error？

File: pkg/snapshot/overlay.go [hunk 4]
```diff
@@ -1170,6 +1179,18 @@ func (o *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
...
 		return "", snapshots.Info{}, errors.Wrap(err, "failed to get snapshot info")
 	}
+	img, ok := info.Labels[label.CRIImageRef]
+	if !ok {
+		img, ok = info.Labels[label.TargetImageRef]
+	}
+	if ok {
+		log.G(ctx).Infof("found imageRef: %s", img)
+		if err := os.WriteFile(filepath.Join(path, "image_ref"), []byte(img), 0644); err != nil {
+			log.G(ctx).Errorf("write imageRef '%s'. path: %s, err: %s", img, filepath.Join(path, "image_ref"), err.Error())
```
Comment (liulanzheng) [inline pkg/snapshot/overlay.go:1189]: use ("%v", err) to print error？

File: go.mod [hunk 5]
```diff
@@ -4,14 +4,17 @@ go 1.19
 
 require (
 	github.com/containerd/containerd v1.6.26
-	github.com/containerd/continuity v0.3.0
-	github.com/containerd/go-cni v1.1.6
+	github.com/containerd/continuity v0.4.3
+	github.com/containerd/go-cni v1.1.9
+	github.com/containerd/log v0.1.0
+	github.com/data-accelerator/zdfs v0.1.3
```
Comment (AkihiroSuda) [inline go.mod:10]: The package lacks license
- https://github.com/data-accelerator/zdfs/issues/2

Comment (BigVan) [reply go.mod:10]: Sorry...
I have added Apache license to the project.

**Pull request merged**